### PR TITLE
Vote log

### PR DESCRIPTION
**Pull request by nanaya**
No interface or anything, just logging who did the vote and record the current votes if the vote causes kudosu grant/revoke.

### CODE REVIEW

File: app/Models/BeatmapDiscussion.php [hunk 1]
```diff
@@ -143,19 +143,25 @@ public function refreshKudosu($event)
...
             if ($event === 'vote') {
                 if ($change > 0) {
                     $beatmapsetEventType = BeatmapsetEvent::KUDOSU_GAIN;
                 } else {
                     $beatmapsetEventType = BeatmapsetEvent::KUDOSU_LOST;
                 }
+
+                $eventExtraData['votes'] = $this
+                    ->beatmapDiscussionVotes()
+                    ->get()
```
Comment (notbakaneko) [inline app/Models/BeatmapDiscussion.php]: `->beatmapDiscussionVotes`  is already a Collection so `()->get()` seems unnecessary?

Comment (nanaya) [reply app/Models/BeatmapDiscussion.php]: apparently not.

```
Undefined property: Illuminate\Database\Eloquent\Relations\HasMany::$map in /home/edho/osu/osu-web/app/Models/BeatmapDiscussion.php:156
```

┐(･_･)┌

**Comment by notbakaneko** [comment]
>_>

**Comment by nanaya** [comment]
>_<

**Pull request merged**